### DISCUSSION TITLE
Post-processing of HTML output

### DISCUSSION THREAD

**Discussion by MinimizationOrg** [quarto-dev/quarto-cli#5462] [Q&A]
I have a program called `postprocess` that, given an input HTML file, produces an output HTML file:
```bash
postprocess input.html > output.html
```
My question is how to introduce this post-processing into the Quarto pipeline. My current solution is to have a bash script `do_post_processing.sh` that searches recursively for all HTML files in a folder and applies `postprocess`, then to call this bash script as a `post-render`:

```yml
project:
  type: website
  post-render: 
    - bash ./do_post_processing.sh _site
 ``` 
However, this is highly inefficient as it processes all the HTML files every time, even if only one of them is new. For example, when using Quarto preview in VS Code, only one HTML file is produced at a time and thus it is not necessary to post-process the others.

Thank you in advance for your help!

(By the way, Quarto preview is quite slow, even if only one *.md has been modified. I'm not sure if this is expected...)

**Comment by mcanouil** [comment]
`post-render` applies to a project, thus to everything.
If you want your script to work only on files that have changed, you'll have to come up with some way to check for date I'm afraid.

> (By the way, Quarto preview is quite slow, even if only one *.md has been modified. I'm not sure if this is expected...)

What does it mean "quite slow"? Compared to what? In which context? Could you elaborate on this?
